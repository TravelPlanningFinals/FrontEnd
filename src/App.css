/*
  Example of using Tailwind classes in a CSS file:
*/
body {
  text-align: center;
  padding: 10px;
  @apply bg-slate-800;
}
