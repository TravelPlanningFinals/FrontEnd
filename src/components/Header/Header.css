.header {
  display: flex;
  justify-content: space-between;
  margin-right: 2vw;
  margin-left: 2vw;
}
