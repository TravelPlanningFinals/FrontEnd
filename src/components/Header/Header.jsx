import React from 'react';
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <div>
      <Link to="/">Back to Home (this is header)</Link>
    </div>
  );
}
