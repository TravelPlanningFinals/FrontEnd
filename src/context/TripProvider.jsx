import { useContext, createContext, useState } from 'react';

const TripContext = createContext();

const TripProvider = ({ children }) => {
  const [trips, setTrips] = useState([]);
  const contextValue = { trips, setTrips };
  return (
    <TripContext.Provider value={contextValue}>{children}</TripContext.Provider>
  );
};

const useTrips = () => {
  const context = useContext(TripContext);

  if (context === undefined) {
    throw new Error('has to be within an TripConext Provider');
  }
  return context;
};

export { TripProvider, useTrips };
