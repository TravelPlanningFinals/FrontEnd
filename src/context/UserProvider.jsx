import { createContext, useContext, useEffect, useState } from 'react';
import fetchUser from '../services/user';
import { getCurrentUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [loading, setLoading] = useState(true);
  const [user, setUser] = useState(null);

  useEffect(() => {
    // fetchUser()
    //   .then((fetchedUser) => {
    //     setUser(fetchedUser);
    //   })
    //   .catch((error) => {
    //     throw new Error(`Error: ${error}`);
    //   });
    getCurrentUser()
      .then((user) => setUser(user))
      .finally(() => setLoading(false));
  }, []);

  const contextValue = { user, loading };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserContext Provider');
  }
  return context;
};
export { UserProvider, useUser };
