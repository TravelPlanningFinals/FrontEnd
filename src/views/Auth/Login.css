.login {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 2px solid #3c493f;
  border-radius: 6px;
  width: 26vw;
  margin-left: 37vw;
  margin-right: 37vw;
  margin-bottom: 15vw;
  align-items: center;
  text-align: center;
}
.button3 {
  border: 2px solid black;
  border-radius: 3px;
  padding: 3px;
  margin: 20px;
}
.signin-message {
  margin: 1vw;
}

img {
  width: 350px;
  height: 350px;
  margin-left: 40.3%;
  display: flex;
  align-items: center;
}
