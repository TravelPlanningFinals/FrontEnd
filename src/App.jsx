import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import './App.css'; /* Global CSS */
// import styles from './App.module.css'; /* CSS Modules */
import Footer from './components/Footer/Footer';
import Header from './components/Header/Header';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import { TripProvider } from './context/TripProvider';
import { UserProvider } from './context/UserProvider';
import AboutUs from './views/About Us/AboutUs';
import Login from './views/Auth/Login';
import PlanPage from './views/Plan/PlanPage';
import Trips from './views/Trips/Trips';

export default function App() {
  return (
    <BrowserRouter>
      <Header />
      <main>
        <UserProvider>
          <TripProvider>
            <Switch>
              {/* <Landing /> */}
              <Route exact path="/">
                <Login />
              </Route>

              <Route exact path="/aboutus">
                <AboutUs />
              </Route>

<<<<<<< HEAD
              {/* <Route path="/login"> */}
              <Route path="/landing">
                {/* <Login /> */}
                <Landing />
              </Route>

              <Route path="/register">
                <Login isSigningUp />
              </Route>

              <Route path="/trips">
                <Trips />
              </Route>

              <PrivateRoute path="/plan">
                <PlanPage />
              </PrivateRoute>

=======
              <Route path="/register">
                <Login isSigningUp />
              </Route>

              <Route path="/trips">
                <Trips />
              </Route>

              <PrivateRoute path="/plan/:id">
                <PlanPage />
              </PrivateRoute>

>>>>>>> 173f3338b263993ed198509f8f39f9583f0b6584
              {/* <Route path="/plan">
              <PlanPage />
            </Route> */}
            </Switch>
          </TripProvider>
        </UserProvider>
      </main>
      <Footer />
    </BrowserRouter>
  );
}
