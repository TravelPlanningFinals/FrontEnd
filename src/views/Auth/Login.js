import React from 'react';
import './Login.css';

export default function Login() {
  return (
    <div className="login">
      <h2 className="signin-message">Sign in with Github OAuth</h2>
      <button
        /*onClick={app.use('/api/v1/github', require('./controllers/github'))}*/
        class="bg-transparent hover:bg-pink-500 text-red-700 font-semibold hover:text-white py-2 px-4 border border-red-500 hover:border-transparent rounded m-6"
      >
        {/* <a href={'https://travelapp70.herokuapp.com/api/v1/github/login'}> */}
        {/* !!!!!this link above auto deploys MAIN branch. We need to activate this one, and disable the one below for presentation !!!!!*/}

        <a href={`${process.env.HEROKU_URL}api/v1/github/login`}>
          {' '}
          {/* !!!! this is dev link!!!!! */}
          Sign in
        </a>
      </button>
    </div>
  );
}
