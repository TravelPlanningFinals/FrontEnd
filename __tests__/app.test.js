import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../src/App';
import Header from '../src/components/Header/Header';
import { TripProvider } from '../src/context/TripProvider';
import { UserProvider } from '../src/context/UserProvider';
import Login from '../src/views/Auth/Login';

test('Renders landing page/login loads', () => {
  const container = render(
    // <BrowserRouter>
    //   <Header>
    //     <main>
    //       <UserProvider>
    //         <TripProvider>
    <App />
    //         </TripProvider>
    //       </UserProvider>
    //     </main>
    //   </Header>
    // </BrowserRouter>
  );

  const signInButton = screen.getByRole('button', { name: /sign in/i });
  const aboutUsLink = screen.getByRole('link', { name: /about us/i });
  const signOut = screen.getByRole('button', { name: /sign out/i });
  const tripsLink = screen.getByRole('link', { name: /back to your trips/i });

  expect(signInButton).toBeInTheDocument();
  expect(aboutUsLink).toBeInTheDocument();
  expect(signOut).toBeInTheDocument();
  expect(tripsLink).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
