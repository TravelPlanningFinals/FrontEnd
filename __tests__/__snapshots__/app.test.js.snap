// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders landing page/login loads 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="back-to-trips"
        >
          <a
            href="/trips"
          >
            Back to your trips
          </a>
        </div>
        <div
          class="sign-out"
        >
          <button>
            Sign Out
          </button>
        </div>
      </div>
      <main>
        <div
          class="login"
        >
          <h2
            class="signin-message"
          >
            Sign in with Github OAuth
          </h2>
          <button
            class="bg-transparent hover:bg-pink-500 text-red-700 font-semibold hover:text-white py-2 px-4 border border-red-500 hover:border-transparent rounded m-6"
          >
            <a
              href="http://localhost:7890/api/v1/github/login"
            >
               
              Sign in
            </a>
          </button>
        </div>
      </main>
      <div
        class="footer"
      >
        <a
          href="/aboutus"
        >
          About Us
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="back-to-trips"
      >
        <a
          href="/trips"
        >
          Back to your trips
        </a>
      </div>
      <div
        class="sign-out"
      >
        <button>
          Sign Out
        </button>
      </div>
    </div>
    <main>
      <div
        class="login"
      >
        <h2
          class="signin-message"
        >
          Sign in with Github OAuth
        </h2>
        <button
          class="bg-transparent hover:bg-pink-500 text-red-700 font-semibold hover:text-white py-2 px-4 border border-red-500 hover:border-transparent rounded m-6"
        >
          <a
            href="http://localhost:7890/api/v1/github/login"
          >
             
            Sign in
          </a>
        </button>
      </div>
    </main>
    <div
      class="footer"
    >
      <a
        href="/aboutus"
      >
        About Us
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
